records/2023-01-26.bounce-rate-automated-emergency-brake.md

```diff
@@ -0,0 +1,83 @@
+![bouncerateformula](./diagrams/2023-01-26.bounce-rate-automated-emergency-brake/bouncerate.png)
+
+### AWS Thresholds
+- A bounce rate is considered **Healthy** when it remains below 5%.
+- A bounce rate is considered **Under review** when it is between 5% and 10% (>=5% and <10%).
+- A bounce rate is considered **At risk** when it is greater than or equal to 10%.
+
+### Why the bounce rate is important
+A high bounce rate often indicates large amounts of unsolicited email are being sent and email providers take this into account when deciding whether to deliver email.  A high bounce rate can cause emails to be sent to users' spam folders and AWS could suspend our email sending cabability if our bounce rate exceeds a certain threshold.
+
+### How GC Notify's bounce rate is measured by AWS
+All services using the GC Notify platform are seen in aggregate by AWS. The GC Notify platform as a whole has a single bounce rate. Individual services contribute to the bounce rate, with higher-volume services having a larger impact than lower-volume services.
+
+## Why this feature is needed
+The purpose of this feature is to keep the bounce rate as low as possible which will mean a positive sending reputation for the GC Notify platform. This will ensure emails are consistently trusted and delivered by email providers.  This can be done by monitoring each service's bounce rate and automatically suspending their service when they reach a certain threshold.  It will also be necessary to provide users with guides to explain these concepts and help to them reduce their bounce rate.
+
+## Proposed solution
+
+### Calculate service bounce rates
+Record the bounce status of every notification and calculate number of emails sent and the bounce rate for each service over the past [bounce-rate-window]
+
+### Notify a service when it is within [bounce-rate-warning-percentage] of the [bounce-rate-threshold]
+When a service is approaching the [bounce-rate-threshold], notify the service owner so they can take action to reduce their bounce rate.  The message should contain a link to the service's dashboard where they can see their bounce rate and take action to reduce it.
+
+#### Minimum volume threshold
+Low-volume senders won't have a large impact on the bounce rate, and notifying them directly may lead to many support calls. In order to limit this, only notify services that are sending more than [bounce-rate-service-volume-minimum] emails in the [bounce-rate-window].
+
+### Suspend a service when it reaches the [bounce-rate-threshold]
+When a service reaches the [bounce-rate-threshold], suspend their service to prevent them from sending any more emails. Notify the service owner that their service has been suspended and provide them with a link to the service's dashboard where they can see their bounce rate and take action to reduce it.
+
+:question: _How to decide when it is appropriate to unsuspend a service?_
```

Comment:
My current take is that this will require a process of a) communicating with users through support to tell them what the issue is and provide a fix on their end first, b) looking at our AWS account bounce rate and determine if we are in a danger zone, c) once users fixed their distribution list, review their fix and wait for Notify bounce rate to go down if in a dangerous zone to reactivate the service, d) monitor the service and their bounce rate in the following days to check that their fix was effective. Would that make sense?